scripts/newinstall.sh

```diff
@@ -386,9 +386,12 @@ miniconda::install() {
 	(
 		set -e
 
-		# the miniconda installer seems to complains if the filename does not end
-		# with .sh
-		tmpfile=$(mktemp -t "XXXXXXXX.${miniconda_file_name}")
+		# XXX the miniconda installer seems to complains if the filename does not
+		# end with .sh
+		#
+		# XXX GNU mktemp will complain about extra Xs in the filename that are too
+		# small of a consecutive sequence to constitute a legal template.
+		tmpfile=$(mktemp -t "XXXXXXXX.${safe_miniconda_file_name/X/_}")
```

Comment:
Should be `miniconda_file_name`?

Should be `${miniconda_file_name//X/_}` (to guard against multiple occurrences of `X` in the filename)?